Exiled.API/Features/Roles/Role.cs

```diff
@@ -185,13 +185,21 @@ protected Role(PlayerRoleBase baseRole)
         /// <inheritdoc/>
         public override int GetHashCode() => base.GetHashCode();
 
+        /// <summary>
+        /// Sets the player's <see cref="RoleTypeId"/>.
+        /// </summary>
+        /// <param name="newRole">The new <see cref="RoleTypeId"/> to be set.</param>
+        /// <param name="reason">The <see cref="Enums.SpawnReason"/> defining why the player's role was changed.</param>
+        public virtual void Set(RoleTypeId newRole, SpawnReason reason = Enums.SpawnReason.ForceClass) =>
+            Set(newRole, reason, RoleSpawnFlags.All);
+
         /// <summary>
         /// Sets the player's <see cref="RoleTypeId"/>.
         /// </summary>
         /// <param name="newRole">The new <see cref="RoleTypeId"/> to be set.</param>
         /// <param name="reason">The <see cref="Enums.SpawnReason"/> defining why the player's role was changed.</param>
         /// <param name="spawnFlags">The <see cref="RoleSpawnFlags"/> defining player spawn logic.</param>
-        public virtual void Set(RoleTypeId newRole, SpawnReason reason = Enums.SpawnReason.ForceClass, RoleSpawnFlags spawnFlags = RoleSpawnFlags.All) =>
+        public virtual void Set(RoleTypeId newRole, SpawnReason reason, RoleSpawnFlags spawnFlags) =>
```

Comment:
Maybe add an overload that doesn't take any params other than RoleTypeId? So plugins aren't required to input the reason property

Reply:
Already, you can use  first overload `Set(RoleTypeId, SpawnReason = SpawnReason.ForceClass)` without second argument

Reply:
Oh I see, my bad